[flake]: TestBatchedAssetIssuance/basic_asset_creation flake

tapgarden struck again with the aforementioned test failing flakily in [this job](https://github.com/lightninglabs/taproot-assets/actions/runs/24229766650/job/70738609921?pr=2053):

```
--- FAIL: TestBatchedAssetIssuance (40.79s)
    --- FAIL: TestBatchedAssetIssuance/basic_asset_creation (4.73s)
        test_sqlite.go:11: Creating new SQLite DB handle for testing: /tmp/TestBatchedAssetIssuancebasic_asset_creation3764323262/001/tmp.db
        planter_test.go:650: 
            	Error Trace:	/home/runner/work/taproot-assets/taproot-assets/tapgarden/planter_test.go:650
            	            				/home/runner/work/taproot-assets/taproot-assets/tapgarden/planter_test.go:604
            	            				/home/runner/work/taproot-assets/taproot-assets/tapgarden/planter_test.go:1195
            	            				/home/runner/work/taproot-assets/taproot-assets/tapgarden/planter_test.go:2123
            	Error:      	"[]" should have 1 item(s), but has 0
            	Test:       	TestBatchedAssetIssuance/basic_asset_creation
    --- FAIL: TestBatchedAssetIssuance/minting_with_cancellation (3.45s)
```

--- a/tapgarden/planter_test.go
+++ b/tapgarden/planter_test.go
@@ -572,11 +572,12 @@ func (t *mintingTestHarness) finalizeBatchAssertFrozen(
 	)
 	require.NoError(t, err)
 
-	// We only want to know if a new batch gets to the frozen state. So the
-	// list of existing batches should only contain the already frozen.
+	// We only want to know if a new batch gets to a frozen-or-later
+	// state. Record all existing batches in those states so the poll
+	// loop can distinguish them from newly frozen batches.
 	var existingFrozenBatches []*tapgarden.MintingBatch
 	fn.ForEach(existingBatches, func(batch *tapgarden.VerboseBatch) {
-		if batch.State() == tapgarden.BatchStateFrozen {
+		if batchFrozenStates.Contains(batch.State()) {
 			existingFrozenBatches = append(
 				existingFrozenBatches, batch.ToMintingBatch(),
 			)
@@ -623,6 +624,8 @@ func (t *mintingTestHarness) assertNewBatchFrozen(
 
 	var newBatches []*tapgarden.MintingBatch
 	err := wait.NoError(func() error {
+		newBatches = nil
+
 		currentBatches, err := t.store.FetchAllBatches(
 			context.Background(),
 		)
@@ -640,7 +643,12 @@ func (t *mintingTestHarness) assertNewBatchFrozen(
 					newBatches = append(newBatches, batch)
 				}
 			}
-			return nil
+
+			if len(newBatches) > 0 {
+				return nil
+			}
+
+			return fmt.Errorf("new batch not yet frozen")
 		}
 
 		return fmt.Errorf("no new batches created")

